src/fetchLabels.ts

```diff
@@ -7,15 +7,36 @@ type Label = {
 
 export default async function fetchLabels(
   token: string,
-  pullRequest: any
+  context: any
 ): Promise<Label[]> {
   const client = github.getOctokit(token);
 
-  const PRPayload = await client.rest.pulls.get({
-    owner: pullRequest.owner,
-    repo: pullRequest.repo,
-    pull_number: pullRequest.number,
-  });
-  const allLabels: Label[] = PRPayload.data.labels as Label[];
+  const payload = await getPayload(client, context);
+  const allLabels: Label[] = payload.data.labels as Label[];
   return allLabels;
 }
+
+async function getPayload(client: any, context: any) {
+  if (context.eventName === "push") {
+    return await client.rest.pulls.get({
+      owner: context.repo.owner,
+      repo: context.repo.repo,
+      pull_number: await getPullNumber(client, context),
+    });
+  }
+  return await client.rest.pulls.get({
+    owner: context.issue.owner,
+    repo: context.issue.repo,
+    pull_number: context.issue.number,
+  });
+}
+
+async function getPullNumber(client: any, context: any) {
+  return (
+    await client.rest.repos.listPullRequestsAssociatedWithCommit({
+      commit_sha: context.sha,
+      owner: context.repo.owner,
+      repo: context.repo.repo,
+    })
+  ).data[0].number;
```

Comment:
`listPullRequests...`, and `data[0]` -> In what scenarios can this return multiple items?

Reply:
If one has branches A and B, with commits `sha1` and `sha1` + `sha2` respectively (so B is created off of A), with both having a PR, and the branch is up-to-date with master so that upon rebase merge (so that the rebase merge doesn't update the `sha`), this can return multiple items.

Seemingly the API is smart enough to associate the squashed sha and merge commit sha with a PR.

Shouldn't really happen in practice but I'll look into ways to further mitigate this. Some rough ideas but not sure they're feasible yet:
* go through all merged commits and pick the pr that is associated with all of them
* ignore open PRs

What do you think?

Reply:
I think could also just fail the run if there's more than expected so don't release anything as major if it's supposed to be patch or minor.